packages/types/lib/IDataSource.ts

```diff
@@ -1,10 +1,14 @@
 import type * as RDF from '@rdfjs/types';
 import type { IActionContext } from './IActionContext';
 
-export type IDataSource = string | RDF.Source | {
```

Comment:
Note that we're removing one case here though, which is probably not what we want:
```
{
  type?: string;
  value: string | RDF.Source;
  context?: IActionContext;
}
```

Reply:
Let's move this old type to a new interface: `IDataSourceExpanded`

Our new interface `IDataSourceSerialized` can then extend this one, so that we can remove the overlapping fields.

Also, the `value` field in here should then also include `IDataSourceSerialized` as possible option.

Reply:
I wanted to be more strict with this interface so I ended up overwriting quite a lot of field. I'm not sure if it's the right approach thought.